### PR TITLE
[char/property] Create component from /utils/src/char_property/

### PR DESCRIPTION
**Pull request by behnam**
We have decided to move abstract char-specific functionalities to a new major component, to allow easier discover and consumptions in the public. This diff creates the major component, `unic-char`, and imports the existing `char_property` implementation under that, under crate name `unic-char-property`.

### CODE REVIEW

File: unic/char/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+[package]
+name = "unic-char"
+version = "0.5.0"
+authors = ["The UNIC Project Developers"]
+repository = "https://github.com/behnam/rust-unic/"
+license = "MIT/Apache-2.0"
+keywords = ["text", "unicode", "utilities"]
+description = "UNIC - Utilities"
+categories = ["text-processing", "parsing", "rendering"]
```
Comment (CAD97) [inline unic/char/Cargo.toml]: `rendering`?

File: unic/char/property/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+[package]
+name = "unic-char-property"
+version = "0.5.0"
+authors = ["The UNIC Project Developers"]
+repository = "https://github.com/behnam/rust-unic/"
+license = "MIT/Apache-2.0"
+keywords = ["text", "unicode", "character-property"]
+description = "UNIC - Unicode Character Database - Age"
+categories = ["text-processing", "parsing", "rendering"]
```
Comment (CAD97) [inline unic/char/property/Cargo.toml]: `rendering`?

File: unic/ucd/age/Cargo.toml [hunk 3]
```diff
@@ -15,5 +15,6 @@ exclude = []
 travis-ci = { repository = "behnam/rust-unic", branch = "master" }
 
 [dependencies]
+unic-char-property = { path = "../../char/property", version = "0.5.0" }
```
Comment (CAD97) [inline unic/ucd/age/Cargo.toml]: Be consistent with the trailing `/` and add one here as well.

File: unic/ucd/bidi/Cargo.toml [hunk 4]
```diff
@@ -15,5 +15,6 @@ exclude = []
 travis-ci = { repository = "behnam/rust-unic", branch = "master" }
 
 [dependencies]
+unic-char-property = { path = "../../char/property", version = "0.5.0" }
```
Comment (CAD97) [inline unic/ucd/bidi/Cargo.toml]: Be consistent with the trailing `/` and add one here as well.

File: unic/ucd/category/Cargo.toml [hunk 5]
```diff
@@ -16,5 +16,6 @@ travis-ci = { repository = "behnam/rust-unic", branch = "master" }
 
 [dependencies]
 matches = "0.1"
+unic-char-property = { path = "../../char/property", version = "0.5.0" }
```
Comment (CAD97) [inline unic/ucd/category/Cargo.toml]: Be consistent with the trailing `/` and add one here as well.

File: unic/ucd/normal/Cargo.toml [hunk 6]
```diff
@@ -15,12 +15,12 @@ exclude = ["tests/conformance_tests.rs"]
 travis-ci = { repository = "behnam/rust-unic", branch = "master" }
 
 [dependencies]
-unic-ucd-core = { path = "../core/", version = "0.5.0" }
+unic-char-property = { path = "../../char/property", version = "0.5.0" }
```
Comment (CAD97) [inline unic/ucd/normal/Cargo.toml]: Be consistent with the trailing `/` and add one here as well.

**Comment by behnam** [comment]
The "rendering" category is here based on the fact that many Unicode algorithms are used for rendering of the text, like line breaking and bidi.

Good catch with the slashes. I'll fix.

**Comment by behnam** [comment]
Oh, I just got your point. Well, I'm not sure about `rendering` for `unic-char*`. I think better drop it for now.

**Comment by behnam** [comment]
bors: r+

**Pull request merged**